src/ast/ast.cpp

```diff
@@ -587,18 +587,67 @@ std::string opstr(const Unop &unop)
   return {}; // unreached
 }
 
-std::string AttachPoint::name(const std::string &attach_target,
-                              const std::string &attach_point) const
+AttachPoint AttachPoint::match_override(const std::string &match) const
+{
+  AttachPoint ap = *this; // copy here
+  switch (probetype(ap.provider)) {
+    case ProbeType::kprobe:
+    case ProbeType::kretprobe:
+      ap.func = match;
+      if (match.find(":") != std::string::npos)
+        ap.target = erase_prefix(ap.func);
+      break;
+    case ProbeType::uprobe:
+    case ProbeType::uretprobe:
+    case ProbeType::kfunc:
+    case ProbeType::kretfunc:
+    case ProbeType::tracepoint:
+      // Tracepoint, uprobe, and k(ret)func probes specify both a target
+      // (category for tracepoints, binary for uprobes, and kernel module
+      // for k(ret)func) and a function name.
+      ap.func = match;
+      ap.target = erase_prefix(ap.func);
+      break;
+    case ProbeType::usdt:
+      // USDT probes specify a target binary path, a provider, and a function
+      // name.
+      ap.func = match;
+      ap.target = erase_prefix(ap.func);
+      ap.ns = erase_prefix(ap.func);
+      break;
+    case ProbeType::watchpoint:
+    case ProbeType::asyncwatchpoint:
+      // Watchpoint probes come with target prefix. Strip the target to get the
+      // function
+      ap.func = match;
+      erase_prefix(ap.func);
+      break;
+    case ProbeType::rawtracepoint:
+      ap.func = match;
+      break;
+    case ProbeType::software:
+    case ProbeType::hardware:
+    case ProbeType::interval:
+    case ProbeType::profile:
+    case ProbeType::special:
+    case ProbeType::iter:
+    case ProbeType::invalid:
```

Comment:
nit: Add a default to the switch that throws or does `LOG(BUG)` to account for any added PropType in the future

Reply:
We normally rely on the compiler to generate warnings/errors when there's a missing switch case. Although it looks like Werror has been turned off in CI since the move from Travis to GitHub actions... we need to get that back on!

Reply:
Ok, I added a `LOG(BUG)` for this PR but if we enable `-Werror`, we should drop it (as well as from other enums) to let compiler statically check for missing switch cases.